src/script/run_tox.sh

```diff
@@ -115,11 +115,20 @@ function main() {
             shift
         fi
         tox_path=$(get_tox_path $test_name)
+        if [ -z "$test_name" ]; then
+            test_name=$(basename $tox_path)
+        fi
         venv_path="$build_dir/$test_name"
     else
         test_name=$(basename $tox_path)
     fi
 
+    if [ "$venv_path" -ef "$build_dir" ]; then
```

Comment:
i am not sure if it's a good idea to make `build_dir` a special case. instead i recommend:

```sh
if [ -e "$venv_path" ]; then
  if [ ! -f ${venv_path}/bin/activate ]; then
    echo "venv_path already exists. bailing out" >& 2
    exit 1
  fi
else
  $source_dir/src/tools/setup-virtualenv.sh ${venv_path}
fi
```